### PR TITLE
Only fail  quantized Lora unload when actually merging

### PR DESCRIPTION
**Pull request by BlackHC**
Lora can only merge parameters when the model is not quantized. However, the error-checking code does not check whether we actually want to merge anything or not.

### CODE REVIEW

**Review by BenjaminBossan — APPROVED** [review]
LGTM, thanks for the fix. Let's have another review by @pacman100 before merging.

**Review by pacman100 — APPROVED** [review]
Thank you @BlackHC for the fix, LGTM! 🤗

**Comment by BlackHC** [comment]
Thank you very much!

**Pull request merged**
